modules/cart/src/Plugin/Block/CartBlock.php

```diff
@@ -0,0 +1,220 @@
+
+/**
+ * @file
+ * Contains \Drupal\commerce_cart\Plugin\Block\CartBlock.
+ */
+
+namespace Drupal\commerce_cart\Plugin\Block;
+
+use Drupal\commerce_cart\CartProviderInterface;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Provides a Shopping cart block.
+ *
+ * @Block(
+ *   id = "cart",
+ *   admin_label = @Translation("Shopping Cart"),
+ *   category = @Translation("Commerce")
+ * )
+ */
+class CartBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * The cart provider.
+   *
+   * @var \Drupal\commerce_cart\CartProvider
+   */
+  protected $cartProvider;
+
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The Drupal account to use for checking for access to block.
+   *
+   * @var \Drupal\Core\Session\AccountInterface.
+   */
+  protected $account;
+
+  /**
+   * Constructs a new BlockContentBlock.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\commerce_cart\CartProviderInterface $cart_provider
+   *   The Cart Provider.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The Entity Type Manager.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, CartProviderInterface $cart_provider, EntityTypeManagerInterface $entity_type_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->cartProvider = $cart_provider;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('commerce_cart.cart_provider'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function defaultConfiguration() {
+    return array(
+      'item_text' => 'items',
+      'cart_link' => TRUE,
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function blockForm($form, FormStateInterface $form_state) {
+    $form['cart_item_text'] = [
+      '#type' => 'textfield',
+      '#title' => $this->t('Items text'),
+      '#default_value' => $this->configuration['item_text'],
+      '#description' => $this->t('This text displays after the product count. Defaults to "items"'),
+    ];
+    $form['cart_cart_link'] = [
```

Comment:
"cart_cart_link" is a mistake :)
Sounds like that should not be configurable? Instead, we'd want to have a setting for turning on/off the dropdown?